iotdb/src/main/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessor.java

```diff
@@ -171,20 +171,27 @@ public boolean write(TSRecord tsRecord) throws BufferWriteProcessorException {
     long memUsage = MemUtils.getRecordSize(tsRecord);
     BasicMemController.UsageLevel level = BasicMemController.getInstance()
         .acquireUsage(this, memUsage);
-    for (DataPoint dataPoint : tsRecord.dataPointList) {
-      workMemTable.write(tsRecord.deviceId, dataPoint.getMeasurementId(), dataPoint.getType(),
-          tsRecord.time,
-          dataPoint.getValue().toString());
-    }
-    valueCount++;
+
     String memory;
     switch (level) {
       case SAFE:
+        for (DataPoint dataPoint : tsRecord.dataPointList) {
```

Comment:
This will cause errors.
We have modified the latest time in filenode processor, so we must write the data to the memtable.

Reply:
I fix this by undoing the time update when the insertion fails.